### PR TITLE
Fix Notebook 04 for invoke_async with context variables

### PR DESCRIPTION
**Pull request by alexchaomander**
### Motivation and Context
New Python API changes the signature for invoke_async.

This aligns it and makes it clearer that the context passed into invoke_async are context variables.

This fixes the current issue where the chat function will just repeat back the user input instead of actually chatting

### CODE REVIEW

File: samples/notebooks/python/04-context-variables-chat.ipynb [hunk 1]
```diff
@@ -238,9 +238,9 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.10.10"
+      "version": "3.10.9"
```
[the remark this answers is not in the document]
Comment (alexchaomander) [reply samples/notebooks/python/04-context-variables-chat.ipynb:241]: Not intentional, it just sometimes happens when I edit in VSCode. Not sure why.

File: samples/notebooks/python/04-context-variables-chat.ipynb [hunk 2]
```diff
@@ -169,16 +169,16 @@
         "async def chat(input_text: str) -> None:\n",
         "    # Save new message in the context variables\n",
         "    print(f\"User: {input_text}\")\n",
-        "    context[\"user_input\"] = input_text\n",
+        "    context_variables[\"user_input\"] = input_text\n",
         "\n",
         "    # Process the user message and get an answer\n",
-        "    answer = await chat_function.invoke_async(context)\n",
+        "    answer = await chat_function.invoke_async(context_variables)\n",
```
Comment (awharrison-28) [inline samples/notebooks/python/04-context-variables-chat.ipynb:175]:
```suggestion
        "    answer = await chat_function.invoke_async(variables=context_variables)\n",
```

**Review by awharrison-28 — APPROVED** [review]

**Comment by dluc** [comment]
is the syntax aligned with the version in requirements.txt?

**Review by lemillermicrosoft — APPROVED** [review]

**Comment by alexchaomander** [comment]
> is the syntax aligned with the version in requirements.txt?
Yes!

**Pull request merged**